mutiny-core/src/auth.rs

```diff
@@ -76,7 +77,11 @@ impl<S: MutinyStorage> MutinyAuthClient<S> {
                 self.retrieve_new_jwt().await?;
                 self.authenticated_request(method, url, body).await
             }
-            _ => Ok(res),
+            StatusCode::OK | StatusCode::ACCEPTED | StatusCode::CREATED => Ok(res),
+            code => Err(MutinyError::Other(anyhow!(
```

Comment:
I don't think we need to anyhow this. We have connection errors.

Reply:
This gives more info, was useful for debugging

Reply:
You could log the error here and then return the connection error.